src/array.vala

```diff
@@ -44,19 +44,25 @@ public class Vast.Array : Object
 
     public size_t size {
         get {
-            if (dimension == 0) {
-                return 0;
-            } else {
-                size_t size = 1;
-                for (var i = 0; i < dimension; i++) {
-                    size *= _shape[i];
-                }
-                return size;
+            size_t size = 1;
+            for (var i = 0; i < dimension; i++) {
+                size *= _shape[i];
             }
+            return size;
         }
     }
 
-    public Bytes? data { get; construct; default = null; }
+    /* Internal bytes that stores the array; cannot be null */
+    public Bytes? data { get; construct;}
+    internal uint8 * _baseptr;
+
+    construct {
+        /* provide a default buffer is nothing allocated */
+        if(_data == null) {
+            _data = new Bytes (new uint8[scalar_size * size]);
+        } 
+        _baseptr = _data.get_data();
+    }
```

Comment:
I don't think any array with  `data == null` is in a valid state; and that's concerning. 

shape and strides are immutable, and I don't think we shall change such fundamental assumptions, just because we need a quick hack in PyGObject. We can probably come up with another way to pass in shape to the constructor for PyGObject.